### ISSUE TITLE
Delay Between Actions

### ISSUE THREAD

**Issue by zoiodelula** [grimaldello/the-crims-italian-bot-v2#2]
There is a way to setup the delay between the actions, I tried to do it but not working, there something that the bot is doing too fast like exiting club when someone inside

**Comment by grimaldello** [comment]
Hi @zoiodelula, thank you for reporting.

At the moment there is a way to increase/decrease or add delay (wait) time between actions, but it is not so intuitive how to do it
If you want to add wait time before exit rave if someone is inside, you can search for `this.logger.info("FOUND VISITOR! Exiting...", Logger_1.LogColor.WARNING);` in release.js file and next to it add `yield this.waitUtils.waitMilliSeconds(1000);`.
You should obtain something like this:
```
else {
    this.logger.info("FOUND VISITOR! Exiting...", Logger_1.LogColor.WARNING);
    yield this.waitUtils.waitMilliSeconds(1000); // to wait 1 second before start the next action
}
```

However, an improvement (not so difficult to implement) is to add delay parameters for all actions in settings (I can implement in these days), so you can change the delay directly there without searching it in code.
I can also add a global delay time (maybe it can be useful)

**Comment by zoiodelula** [comment]
public async waitMilliSeconds(min: number, max: number) {
        const randomNumber: number = Math.random() * (max - min) + min;
        return await new Promise( resolve => setTimeout(resolve, randomNumber));
        
        i changed this part of code and works

**Comment by zoiodelula** [comment]
Getting stuck now with this error
VM364:4020 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'x')
    at LinearDOMCoordinatePathFinderStrategy.findPath (<anonymous>:4020:37)
    at BotDOMHelper.<anonymous> (<anonymous>:3703:62)
    at Generator.next (<anonymous>)
    at <anonymous>:3611:71
    at new Promise (<anonymous>)
    at __awaiter (<anonymous>:3607:12)
    at BotDOMHelper.moveFromStartCoordinateToEndCoordinateAndClick (<anonymous>:3702:16)
    at BotDOMHelper.<anonymous> (<anonymous>:3646:24)
    at Generator.next (<anonymous>)
    at <anonymous>:3611:71

**Comment by grimaldello** [comment]
What I suggest to you is only a temporary change that you can make in `release.js` file in order to perform some tests.

To make the change permanent, it is needed to do it in the source code.

In these days I'm going to add in the Settings, option to set various delay/sleep time before actions (and maybe a global one)

**Comment by zoiodelula** [comment]
Yes Im working on the source and recompiling, got some good results with that code that ive sended, the error that I talk before is related to exit button, i think sometimes the id change. I have a idea to solve that, i remember, in python, i use the text inside the button to solve that

**Comment by grimaldello** [comment]
I'm adding the options for the delays after actions in recharge routine.

Checking if it's correctly working and then push and merge in master.

Will notify you when pushed.

**Comment by zoiodelula** [comment]
Okay bro, and can you accept me on discord? I think it is better to talk there

**Comment by grimaldello** [comment]
Released the new version 0.0.4.

You can find changes and new settings in this commit: [Commit changes](https://github.com/grimaldello/the-crims-italian-bot-v2/commit/37e33862fe905a923debb7a8a2bee0ebe63faa27)
